lib/smartsheet/client_behaviour.ex

```diff
@@ -0,0 +1,17 @@
+defmodule Smartsheet.ClientBehaviour do
+  @callback get_sheet(String.t(), List) ::
+              {:ok, Smartsheet.Response.t(), Smartsheet.Sheet.t()}
+              | {:error, Smartsheet.Response.t()}
+
+  @callback create_sheet(map()) ::
+              {:ok, Smartsheet.Response.t(), Smartsheet.Sheet.t()}
+              | {:error, Smartsheet.Response.t()}
+
+  @callback add_rows(String.t(), any) ::
+              {:ok, Smartsheet.Response.t(), [Smartsheet.Row.t()]}
+              | {:error, Smartsheet.Response.t()}
+
+  @callback update_rows(String.t(), any) ::
+              {:ok, Smartsheet.Response.t(), [Smartsheet.Row.t()]}
+              | {:error, Smartsheet.Response.t()}
```

Comment:
i tried to get an error / failure / warning by putting nonsense in these definitions, like
```elixir
:: {:no_thanks}
```
and it didn't warn me at all. so idk if i'm missing something?

Reply:
Ah yeah I think you'll have to run `dialyzer` to get warnings on the return types. The behaviour only checks the function signatures.